src/exec/mod.rs

```diff
@@ -125,6 +193,14 @@ enum ArgumentTemplate {
 }
 
 impl ArgumentTemplate {
+    pub fn has_tokens(&self) -> bool {
+        if let ArgumentTemplate::Tokens(_) = self {
```

Comment:
if we are not capturing anything, maybe use `match` instead? Probably just a matter of taste..